source/dub/init.d

```diff
@@ -168,6 +168,8 @@ private string getUserName()
 		import core.sys.posix.pwd, core.sys.posix.unistd, core.stdc.string : strlen;
 		import std.algorithm : splitter;
 
+		// Bionic doesn't have pw_gecos on ARM
```

Comment:
I think the critical place is here: https://github.com/android/platform_bionic/blob/5725481f1e4cb2e2042b5469b5863023556cf2dd/libc/include/pwd.h#L68

The `pw_gecos` field is missing for non LP64.